### PR TITLE
Improving test suite Issue 94

### LINKED ISSUE #94: Aumentar confiabilidade da suíte de testes

### PR DESCRIPTION
**Pull request by erickzanardo**
This PR closes #94 

### How was it before?

- the project test suit had a lot of breaking specs

### What has changed?

- specs was fixes, or removed (when they did not make sense anymore) and now the test suite is now passing

### What should I pay attention when reviewing this PR?

Nothing special

### Is this PR dangerous?

kind of, a lot of code was changed or removed

### CODE REVIEW

File: app/controllers/admin/cycles_controller.rb [hunk 1]
```diff
@@ -72,6 +72,13 @@ def new
 
   def create
     @cycle = Cycle.new cycle_params
+
+    if @cycle.phases.empty?
+      @cycle.errors[:phases] << "não pode ficar em branco"
```
Comment (guimello) [inline app/controllers/admin/cycles_controller.rb]: you can use `I18n.t` here.

File: app/models/phase.rb [hunk 2]
```diff
@@ -58,11 +58,12 @@ def finished?
   end
 
   def in_progress?
-    ((self.initial_date <= Time.zone.now) and (self.final_date >= Time.zone.now)) && self.plugin_relation.ready?
+    ((self.initial_date <= Time.zone.now) and (self.final_date >= Time.zone.now)) &&
+      (self.plugin_relation.present? && self.plugin_relation.ready?)
```
Comment (guimello) [inline app/models/phase.rb]: IMO we should not check plugin_relation existence. If we don't hava a plugin relation, something is very wrong with the system.

File: spec/models/cycle_spec.rb [hunk 3]
```diff
@@ -31,8 +31,10 @@
   include_examples 'paranoia'
 
   # PRESENCE
-  [:name, :subdomain, :title, :about, :initial_date, :final_date, :color, :description].each do |attr|
-    it  "should validate the presence of #{attr} " do
+  # the name property is also required and has a validation, but as seen on Cycle#set_name, the name is setted with the value of the title
```
Comment (guimello) [inline spec/models/cycle_spec.rb]: typo `set`

**Review by guimello — CHANGES_REQUESTED** [review]

**Review by marcoamarelo — APPROVED** [review]
PR accepted!

**Review by guimello — APPROVED** [review]
🚀

**Pull request merged**